File: mockmate-backend/src/main/resources/runners/python/runner_int.py
import sys

{{USER_CODE}}

lines = sys.stdin.read().strip().split('\n')
try:
    n = int(lines[0].strip())
except Exception as e:
    print(f"Invalid integer: {e}", file=sys.stderr)
    sys.exit(1)
try:
    sol = Solution()
    if not hasattr(sol, "{{methodSignature}}"):
        print("method {{methodSignature}} not found", file=sys.stderr)
        sys.exit(1)
    result = sol.{{methodSignature}}(n)
except Exception as e:
    print(f"Error: {e}", file=sys.stderr)
    sys.exit(1)
if isinstance(result, bool):
    print(str(result).lower())
else:
    print(result)

File: mockmate-backend/src/main/resources/runners/java/runner_int.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        try {
            String line = sc.hasNextLine() ? sc.nextLine() : "0";
            int n = parseSingleInt(line);

            Solution sol = new Solution();
            Object result = sol.{{methodSignature}}(n);
            if (result instanceof Boolean) {
                System.out.println(result.toString().toLowerCase());
            } else {
                System.out.println(result);
            }
        } catch (Exception e) {
            e.printStackTrace();
            System.exit(1);
        } finally {
            sc.close();
        }
    }

    static int parseSingleInt(String line) {
        if (line.contains("=")) line = line.substring(line.indexOf("=") + 1);
        line = line.replace(",", "").trim();
        return Integer.parseInt(line);
    }
}

{{USER_CODE}}
